ml-agents/mlagents/trainers/trainer.py

```diff
@@ -139,6 +130,12 @@ def update_policy(self):
         """
         raise UnityTrainerException("The update_model method was not implemented.")
 
+    def curriculum_lesson_changed(self) -> None:
```

Comment:
Clarifying question: Does the idea of a curriculum make sense for all types of `Trainer`? E.g. `BCTrainer`. I am wondering if this method should be specific to `PPOTrainer` (s)?

cc @vincentpierre

Reply:
Yeah, I was rethinking this one recently as well.  I think curricula don't make much sense for `BCTrainer`, though in theory there's no reason why the idea of changing the reset parameters based on certain criteria would be a problem.

Maybe the idea of the `reward_buffer`, which is really the issue here, should be moved up into the TrainerController / Curriculum class since it's only used for that purpose at this time.

Reply:
After some thought, I am going to remove this method and revert the code around the `reward_buffer` to how it was before.  I think we should deal with that issue separately from this PR.